benchmarking/compute_probability_of_improvement.py

```diff
@@ -0,0 +1,251 @@
+                f"The baseline run directory [{args.baseline_runs_dir}] contains runs "
+                f"for the algorithms [{', '.join(baseline_algos)}]. "
+                f"You specified [{args.baseline_algo}], for which no runs can be found"
+                f" in the baseline run directory",
+            )
+
+        if (args.algo == args.baseline_algo) and (
+            args.runs_dir == args.baseline_runs_dir
+        ):
+            warnings.warn(
+                "You are comparing two equal sets of runs. "
+                "This is probably not what you want.",
+            )
+
+        envs = runs_by_algo_and_env[args.algo].keys()
+        baseline_envs = baseline_runs_by_algo_and_env[args.baseline_algo].keys()
+
+        comparison_envs = set(envs).intersection(set(baseline_envs))
+
+        if len(comparison_envs) == 0:
+            raise ValueError(
+                f"The baseline runs are for the environments "
+                f"[{', '.join(baseline_envs)}], while the runs are for the "
+                f"environments [{', '.join(envs)}]. "
+                f"There is no overlap in the environments of the two run sets, so no "
+                f"comparison can be made",
+            )
+
+        ignoring_some_envs = len(comparison_envs) < len(envs)
+        ignoring_some_baseline_envs = len(comparison_envs) < len(baseline_envs)
+        if ignoring_some_envs or ignoring_some_baseline_envs:
+            warnings.warn(
+                f"The baseline runs are for the environments "
+                f"[{', '.join(baseline_envs)}], "
+                f"while the runs are for the environments [{', '.join(envs)}]. "
+                f"The comparison will only be made for the environments "
+                f"[{', '.join(comparison_envs)}].",
+            )
+
+    except ValueError as e:
+        print(e)
+        sys.exit(1)
+
+    (
+        probability_of_improvement,
+        error_interval,
+        n_samples,
+        n_baseline_samples,
+    ) = compute_probability_of_improvement(
+        runs_by_env=runs_by_algo_and_env[args.algo],
+        baseline_runs_by_env=baseline_runs_by_algo_and_env[args.baseline_algo],
+        reps=args.bootstrap_reps,
+    )
+
+    show_path = args.algo == args.baseline_algo
+    algo_str = f"{args.algo} ({args.runs_dir})" if show_path else args.algo
+    baseline_algo_str = (
+        f"{args.baseline_algo} ({args.baseline_runs_dir})"
+        if show_path
+        else args.baseline_algo
+    )
+
+    print(
+        f"Comparison based on {n_samples} samples per environment for {algo_str} and"
+        f" {n_baseline_samples} samples per environment for {baseline_algo_str}.",
+    )
+    print(f"Samples taken in {', '.join(comparison_envs)}")
+    print()
+    print(f"Probability of improvement of {algo_str} over {baseline_algo_str}:")
+    print(
+        f"{probability_of_improvement:.3f} "
+        f"({error_interval[0]:.3f}, {error_interval[1]:.3f}, "
+        f"reps={args.bootstrap_reps})",
```

Comment:
```suggestion
        f"reps={args.bootstrap_reps:,})",
```

Comma-separated b/c this seems to be a big-ish number.

Reply:
Fixed, good point!